### PR TITLE
Feature/sanitizing causes data

### PR DESCRIPTION
**Pull request by elhaw**
We collaborated together to figure out how to sanitize our application state before saving it in redux.

###  Our approach
  

1. Declaring a function that takes the with a parameter named `data` and returns an object that contains the needed data only.
 
2. Calling The function inside `causesReducer`  and passing the function argument `payload` which returned the data we need.

### CODE REVIEW

File: src/store/reducers/causes.js [hunk 1]
```diff
@@ -1,15 +1,49 @@
...
+  else {
+    return null
+  }
+};
 
 export const causesReducer = (state = initialState, { type, payload }) => {
   switch (type) {
     case ACTION_TYPES.CAUSES.SET_DATA:
-      const newState = { ...state, data: payload };
+      let sanitizeData = dataSanitization(payload);
```
Comment (Nilegfx) [inline src/store/reducers/causes.js:44]: this logic shouldn't be in the reducer, it should be in the action (before dispatching it)

File: src/store/actions/causes.js [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+          imgURL,
+          id
+        };
+      }
+    );
+    return {
+      id,
+      causes: sanitizedCauses,
+      causes_heading: {
+        heading_primary,
```
Comment (muubar) [inline src/store/actions/causes.js:27]: I'm noticing that some fields are using snake_case naming convention, while others are using camelCase, can we agree on a single convention?

File: src/store/actions/causes.js [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+        heading_secondary
+      }
+    };
+  }
+  else {
+    return data
```
Comment (muubar) [inline src/store/actions/causes.js:33]: if there's no data, you're returning the data, this is weird, I think we should think about returning something that we can actually control the value of, like `null` for example

Comment (Nilegfx) [reply src/store/actions/causes.js:33]: he was returning null and giving the nature of this function (only sanitize if there is a data) I asked him to return the exact same falsely datatype as is.

Comment (Nilegfx) [reply src/store/actions/causes.js:33]: this doesn't necessarily means that I am correct. i am telling why they made it like this.

Comment (elhaw) [reply src/store/actions/causes.js:33]: > if there's no data, you're returning the data, this is weird, I think we should think about returning something that we can actually control the value of, like `null` for example

The initial value of data is `null`  so if `data` is falsy the function will return data which is `null`

File: src/store/actions/causes.js (hunk 2 above)
Comment (elhaw) [reply src/store/actions/causes.js:27]: We do not explicitly follow a style guide in naming conventios. 
I will consider it. Thanks 👍

**Pull request merged**